src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -882,11 +890,13 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             return (elementNames.ToImmutable(), builder.ToImmutableAndFree());
         }
 
-        private static void RemoveDuplicateInferredTupleNames(ArrayBuilder<string> inferredElementNames, HashSet<string> uniqueFieldNames)
+        // Returns true if all inferred names were removed.
+        // Returns false otherwise.
+        private static bool RemoveDuplicateInferredTupleNamesAndReportEmptied(ArrayBuilder<string> inferredElementNames, HashSet<string> uniqueFieldNames)
         {
             if (inferredElementNames == null)
             {
-                return;
+                return false;
```

Comment:
>return false; [](start = 16, length = 13)

Should we return true here?

Reply:
I could adjust to work that way if you feel it's clearer. But my thinking here is that the input array builder was not emptied, since you didn't pass one.